sigmastate/src/main/scala/sigmastate/trees.scala

```diff
@@ -16,13 +16,29 @@ import sigmastate.utxo.{Transformer, SimpleTransformerCompanion}
 import scala.collection.mutable
 import scala.collection.mutable.ArrayBuffer
 
+
+/**
+  * Basic trait for inner nodes of crypto-trees, so AND/OR/THRESHOLD sigma-protocol connectives
+  */
+trait SigmaConjecture extends SigmaBoolean {
+  val children: Seq[SigmaBoolean]
```

Comment:
Let's avoid using vals in new traits (legacy code aside) and use defs instead.

Reply:
done